cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -270,6 +287,7 @@ void XDMFFile::write_meshtags(const mesh::MeshTags<std::int32_t>& meshtags,
 //-----------------------------------------------------------------------------
 mesh::MeshTags<std::int32_t>
 XDMFFile::read_meshtags(const std::shared_ptr<const mesh::Mesh>& mesh,
+                        const fem::CoordinateElement& entity_element,
```

Comment:
Why is a `CoordinateElement` required here?

Reply:
It is needed here because it is needed below in `extract_local_entities`. The reason is to be able to tell which nodes that define the entity are the ones associated with vertices. I am throwing away non-vertex nodes at the very beginning since these are not needed and parallel communication sends fewer data.

I agree I can avoid this argument if I am able to tell which nodes on a cell sub-entity are associated with vertices. One approach I had in mind was to first ask which nodes (geometry dofs) for the whole cell are associated with vertices. Then ask which dofs live on the entity of given subdimension and then find their intersection. What do you think @garth-wells ?

Reply:
Can you use the `CoordinateElement` that is associated with the Mesh? Otherwise I'd suggest passing just an array that picks out the vertices so we avoid mixing up geometry and topology as far as possible.